lib/pages/Notifications.dart

```diff
@@ -1,143 +1,191 @@
 import 'dart:ui';
+import 'package:carriage_rider/models/Ride.dart';
 import 'package:carriage_rider/pages/RidePage.dart';
+import 'package:carriage_rider/providers/RidesProvider.dart';
+import 'package:carriage_rider/widgets/ScheduleBar.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:carriage_rider/utils/CarriageTheme.dart';
+import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
 
-class Notifications extends StatefulWidget {
-  Notifications({Key key}) : super(key: key);
+DriverNotification driverArrivedNotif(Ride ride, DateTime notifTime) {
+  return DriverNotification(ride, notifTime, 'Your driver is here! Meet your driver at the pickup point for ${ride.endLocation}.');
+}
+DriverNotification driverOnTheWayNotif(Ride ride, DateTime notifTime) {
+  return DriverNotification(ride, notifTime, 'Your driver is on the way! Wait outside at the pickup point for ${ride.endLocation}.');
+}
+DispatcherNotification driverCancelledNotif(Ride ride, DateTime notifTime) {
+  return DispatcherNotification(ride, notifTime, 'Your driver cancelled the ride because they were unable to find you.', Colors.red, Icons.close);
+}
+DispatcherNotification rideEditedNotif(Ride ride, DateTime notifTime) {
+  return DispatcherNotification(ride, notifTime, 'The information for your ride on ${DateFormat('MM/dd/yyyy').format(ride.startTime)} has been edited. Please review your ride info.', Colors.red, Icons.edit);
+}
+DispatcherNotification rideConfirmedNotif(Ride ride, DateTime notifTime) {
+  return DispatcherNotification(ride, notifTime, 'Your ride on ${DateFormat('MM/dd/yyyy').format(ride.startTime)} has been confirmed.', Colors.green, Icons.check);
+}
```

Comment:
might want to coordinate with michael about this, but the notif sender and body text will be handled by the server (and possibly the notification type and creation timestamp too), so the frontend just needs to use that info rather than generating it itself

Reply:
These are more meant for the Notifications page than the live push notifications, because I think those might only take text rather than widgets? Tony is working on that part though so I'm not sure. For the Notifications page, I think I'd just be fetching from backend where hopefully we'll have the rides and times that correspond to the past notifications?

Reply:
mm yeah ill have to think about the flow some more, but the push and in app notifications should be displaying the same text, so at least the sender and body text should already be provided. i dont think michael has started with storing user notifs, so might still be good to let him know